### ISSUE TITLE
macros introduce both define and define-syntax

### ISSUE THREAD

**Issue by LuKuangChen** [mflatt/plait#21]
Why the following program doesn't work?

```racket
#lang plait

(define-syntax (define-1st-order-function stx)
  (syntax-case stx ()
    [(_ (rator rand ...) body)
     #'(splice
        (define (F rand ...) body)
        (define-syntax rator
          (syntax-rules ()
            [(_ rand ...)
             (F rand ...)])))]))

(define-1st-order-function (sum n1 n2)
  (+ n1 n2))
```

```
❌ rename-out: no binding for identifier in: F
```

**Comment by mflatt** [comment]
Thanks for the report! There were a number of problems with definition-generating macros, and I'll push a repair.

**Comment by LuKuangChen** [comment]
I tried to run the same program with the latest plait. `(sum 2 3)` and `sum` work as expected in REPL. But if I put `(sum 2 3)` in the editor, I get an error:

```
❌ F: cannot use identifier tainted by macro transformation in: F
```

**Comment by mflatt** [comment]
Hm... I see that in Racket v8.2, too. It turns out that the implementation is relying on changes that are part of the upcoming v8.3 release — a surprise, and that should be rare, but I can see how that might happen in this case. Also, I'm seeing some installation errors.

I'll reopen and see about adjustments for v8.2 and earlier.

**Comment by mflatt** [comment]
The problem was more straightforward than I initially thought (v8.3 gets rids of syntax tainting, which is otherwise a small obstacle here), and it should be fixed by b628372b4b6c1d10f0805e9ac6bf4e648bd6b0ef.
